src/transformers/models/chameleon/convert_chameleon_weights_to_hf.py

```diff
@@ -0,0 +1,396 @@
+                [loaded[i][f"layers.{layer_i}.feed_forward.w2.weight"] for i in range(num_shards)], dim=1
+            )
+            state_dict[f"model.layers.{layer_i}.mlp.up_proj.weight"] = torch.cat(
+                [loaded[i][f"layers.{layer_i}.feed_forward.w3.weight"] for i in range(num_shards)], dim=0
+            )
+
+    if num_shards == 1:
+        # Unsharded
+        state_dict.update(
+            {
+                "model.embed_tokens.weight": loaded["tok_embeddings.weight"],
+                "model.norm.weight": loaded["norm.weight"],
+                "lm_head.weight": loaded["output.weight"],
+            }
+        )
+    else:
+        state_dict.update(
+            {
+                "model.embed_tokens.weight": torch.cat(
+                    [loaded[i]["tok_embeddings.weight"] for i in range(num_shards)], dim=1
+                ),
+                "model.norm.weight": torch.stack([loaded[i]["norm.weight"] for i in range(num_shards)]).mean(dim=0),
+                "lm_head.weight": torch.cat([loaded[i]["output.weight"] for i in range(num_shards)], dim=0),
+            }
+        )
+
+    # Load VQGAN weights
+    vqgan_path = os.path.join(input_base_path, "tokenizer/vqgan.ckpt")
+    vqgan_state_dict = torch.load(vqgan_path, map_location="cpu")["state_dict"]
+    for k, v in vqgan_state_dict.items():
+        if "decoder" in k:
+            continue  # we dont do image generation yet
+        state_dict[f"model.vqmodel.{k}"] = v
+
+    # Write configs
+    ffn_dim_multiplier = params["ffn_dim_multiplier"] if "ffn_dim_multiplier" in params else 1
+    multiple_of = params["multiple_of"] if "multiple_of" in params else 256
+
+    with open(os.path.join(input_base_path, "tokenizer/text_tokenizer.json")) as tokenizer_file:
+        tokenizer_config = json.load(tokenizer_file)
+        vocabulary_map = tokenizer_config["model"]["vocab"]
+        vocabulary_map["<image>"] = vocabulary_map[
+            "<reserved08707>"
+        ]  # use a reserved token instead of adding a new one
+        del vocabulary_map["<reserved08707>"]
+
+        for token in tokenizer_config["added_tokens"]:
+            if token["content"] == "<reserved08707>":
+                token["content"] = "<image>"
+
+    with open(os.path.join(input_base_path, "tokenizer/text_tokenizer_modified.json"), "w") as f:
+        json.dump(tokenizer_config, f)  # save the new file to init a tokenizer later
+
+    vq_keys_to_replace = [
+        ("ch", "base_channels"),
+        ("out_ch", "out_channels"),
+        ("n_embed", "num_embeddings"),
+        ("ch_mult", "channel_multiplier"),
+    ]
+    with open(os.path.join(input_base_path, "tokenizer/vqgan.yaml")) as vqgan_cfg_file:
+        vq_config = yaml.safe_load(vqgan_cfg_file)["model"]["params"]
+        for old, new in vq_keys_to_replace:
+            vq_config[new] = vq_config[old]
+        vq_config.update(**vq_config["ddconfig"])
+        del vq_config["ddconfig"]
```

Comment:
Needed this else things broke: 
```suggestion
        vq_config = yaml.safe_load(vqgan_cfg_file)["model"]["params"]
        vq_config.update(**vq_config["ddconfig"])
        for old, new in vq_keys_to_replace:
            vq_config[new] = vq_config[old]
        del vq_config["ddconfig"]
```

Reply:
Yep, didn't push some changes yesterday, now it's fixed and the overall conversion works!